### ISSUE TITLE
[Bug] Package details page on site shows 404 before rendering

### ISSUE THREAD

**Issue by Daniel15** [yarnpkg/berry#803]
- [x] I'd be willing to implement a fix

**Describe the bug**

When you hit a package permalink on the new site, it shows a 404 error briefly, before rendering the correct page

**To Reproduce**

Go to https://yarnpkg.com/package/react

**Screenshots**

First shows:
![image](https://user-images.githubusercontent.com/91933/73202539-1d87f700-40f0-11ea-9431-ec1007d53412.png)

But then a bit later, shows the correct page:
![image](https://user-images.githubusercontent.com/91933/73202565-2678c880-40f0-11ea-9d33-ec491d8c80f3.png)

**Comment by Daniel15** [comment]
cc @Haroenv

**Comment by andreialecu** [comment]
To add to this, when going to the permalink, the search bar is also missing.

When searching from the home page it looks like this:

![image](https://user-images.githubusercontent.com/697707/73208902-e9dcb980-414f-11ea-9986-3041fe4c0db4.png)

**Comment by Haroenv** [comment]
This is something to do with the Gatsby setup. Locally it did work, but online this happens indeed. If anyone's intimately knowing dynamic routes in Gatsby, that will help!

**Comment by Haroenv** [comment]
The css also seems to be wrong if you go to a page via the URL, instead of a navigation on the site:

<img width="1440" alt="Screenshot 2020-01-28 at 09 44 43" src="https://user-images.githubusercontent.com/6270048/73248283-f21a1080-41b2-11ea-95df-b6bbe13f0903.png">

**Comment by 7rulnik** [comment]
It would be cool to show package name in a page title so it will be easier to navigate between many tabs with packages.

**Comment by 7rulnik** [comment]
I sent PR for showing a package name in the title.

The main problem in server settings. I tried to run locally with `yarn serve` and got this:

![image](https://user-images.githubusercontent.com/5969049/82740900-a7ba3200-9d55-11ea-85d6-6f18c6765c86.png)

![image](https://user-images.githubusercontent.com/5969049/82740928-e2bc6580-9d55-11ea-94c1-c3e72e185d7b.png)

It works with `gatsby serve` because it uses express. It reads `.cache/match-paths.json` and [serves](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/commands/serve.ts#L55) page that corresponds dynamic path.

`match-paths.json` content is:

```json
[
    {
        "path": "/package",
        "matchPath": "/package/*"
    }
]
```

I added the rewrite for Netlify in #1396, but not sure that it will work for yarnpkg.com. Is it on netlify too? @arcanis
